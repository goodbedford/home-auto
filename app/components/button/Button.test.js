/* eslint-disable no-undef */

import React from "react";
import {shallow} from "enzyme";
import Button from "./Button";
import renderer from "react-test-renderer";


describe("<Button />", () => {
  it("Button snapshot test", () => {
    const component = renderer.create(<Button />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Button should correct default text", () => {
    const component = renderer.create(<Button />);
    const tree = component.toJSON();
    expect(tree.children.length).toBe(1);
    expect(tree.children[0]).toBe("click");
  });
  it("Button should correct text when given props", () => {
    const component = renderer.create(<Button txt="enter" />);
    const tree = component.toJSON();
    expect(tree.children.length).toBe(1);
    expect(tree.children[0]).toBe("enter");
  });
});
