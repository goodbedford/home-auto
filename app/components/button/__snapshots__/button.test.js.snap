// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Button snapshot test 1`] = `
<button
  className="btn"
>
  click
</button>
`;
